backend/cfg/llvmize.ml

```diff
@@ -518,29 +541,64 @@ let data ds =
   let t = get_current_compilation_unit "data" in
   t.data <- List.append t.data ds
 
+(* Define menitoned but not declared data items as extern *)
+let emit_data_extern t =
+  let defined_function_symbols =
+    List.filter_map
+      (fun (d : Cmm.data_item) ->
+        match d with
+        | Cdefine_symbol { sym_name; sym_global = _ } -> Some sym_name
+        | Cint _ | Csymbol_address _ | Cint8 _ | Cint16 _ | Cint32 _ | Csingle _
+        | Cdouble _ | Cvec128 _ | Cvec256 _ | Cvec512 _ | Csymbol_offset _
+        | Cstring _ | Cskip _ | Calign _ ->
+          None)
+      t.data
+    |> String.Set.of_list
+    |> String.Set.union t.defined_function_symbols
+  in
+  let referenced_symbols =
+    List.filter_map
+      (fun (d : Cmm.data_item) ->
+        match d with
+        | Csymbol_address { sym_name; _ } -> Some sym_name
+        | Cdefine_symbol _ | Cint _ | Cint8 _ | Cint16 _ | Cint32 _ | Csingle _
+        | Cdouble _ | Cvec128 _ | Cvec256 _ | Cvec512 _ | Csymbol_offset _
+        | Cstring _ | Cskip _ | Calign _ ->
+          None)
+      t.data
+    |> String.Set.of_list
+    |> String.Set.union t.referenced_symbols
+  in
```

Comment:
Can you try to merge these two matches with the one below that already traverses t.data? Looks like you can just add the symbols to the right sets at line L587 and L591.